File: 코딩 기초 트레이닝/문자열 묶기-p181855/문자열 묶기.js
// https://school.programmers.co.kr/learn/courses/30/lessons/181855

function solution(strArr) {
    const dic = {}
    const answer = [];

    strArr.forEach((s) => {
        if(dic[s.length] === undefined) {
            dic[s.length] = 1;
        } else {
            dic[s.length] += 1;
        }
    });

    for (let key in dic ){
        answer.push(dic[key]);
    }

    return Math.max(...answer);

}

File: 코딩 기초 트레이닝/문자열 묶기-p181855/문자열 묶기.py
# https://school.programmers.co.kr/learn/courses/30/lessons/181855

def solution(strArr):
    dic = {}
    answer = []

    for s in strArr:
        if not len(s) in dic:
            dic[len(s)] = 1
        else:
            dic[len(s)] += 1

    for k in dic:
        answer.append(dic[k])

    return max(answer)
